lib/new_relic/agent/instrumentation/active_support.rb

```diff
@@ -2,11 +2,22 @@
 # See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
 # frozen_string_literal: true
 
-# This is a helper file that will allow apps using ActiveSupport without Rails
-# to still leverage all ActiveSupport based instrumentation functionality
-# offered by the agent that would otherwise be gated by the detection of Rails.
+require 'new_relic/agent/instrumentation/active_support_subscriber'
 
-# ActiveSupport notifications custom events
-if !defined?(Rails) && defined?(ActiveSupport::Notifications) && defined?(ActiveSupport::IsolatedExecutionState)
-  require_relative 'rails_notifications/custom_events'
+DependencyDetection.defer do
+  named :active_support
+
+  depends_on do
+    defined?(ActiveSupport) &&
+      !NewRelic::Agent::Instrumentation::ActiveSupportSubscriber.subscribed?
+  end
+
+  executes do
+    NewRelic::Agent.logger.info('Installing ActiveSupport instrumentation')
+  end
+
+  executes do
+    ActiveSupport::Notifications.subscribe(/\.active_support$/,
```

Comment:
I'm still not clear on why we use `^` and `$` in some places and `\A` and `\z` in others for these patterns.

Reply:
Great question. I think it has more to do with the person's preference at the time of writing. For me, it was copy-pasta. I wonder if one of them has a performance benefit. Do you know off the top of your head? If not, perhaps we can investigate as part of the post-milestone refactor.

Reply:
Good call on post-milestone. I think `\A` and `\z` are best. They represent the absolute beginning and end of the entire text, whereas `^` and `$` represent the start and stop of any given line.

For example, this text blob will be matched with your current regex, but arguably shouldn't.

```text
This is a simple
.active_support
example
```

Given that the input is expected to a) not be user crafted and b) always exactly one line, this regex and others like it should be fine for now.